src/helpers/misc.ts

```diff
@@ -313,6 +313,27 @@ export function debounce<T extends (...args: any) => void>(
   return debounced as DebouncedFunction<T>;
 }
 
+/**
+ * Creates a batched version of a callback so that all calls to it in the same
+ * microtick will only call the original callback once.
+ *
+ * @param callback the callback to batch
+ * @returns a batched version of the original callback
+ *
+ * Copied from odoo/owl repo.
+ */
+export function batched(callback: () => void): () => void {
+  let scheduled = false;
+  return async (...args) => {
+    if (!scheduled) {
+      scheduled = true;
+      await Promise.resolve();
+      scheduled = false;
+      callback(...args);
+    }
+  };
+}
+
```

Comment:
shouldn't batched be linked to the nextanimationframe instead of the microtick ?

Reply:
I pushed a commit but it's :nauseated_face: just because we don't have a direct access to owl's `Scheduler` (same trick used in the test).
I'll go and talk to the framework-js team tomorrow to check if they can export `Scheduler` from owl